modules/system/defaults-write.nix

```diff
@@ -9,6 +9,10 @@ let
     "defaults write ${domain} '${key}' $'${strings.escape [ "'" ] (generators.toPlist { } value)}'";
 
   defaultsToList = domain: attrs: mapAttrsToList (writeDefault domain) (filterAttrs (n: v: v != null) attrs);
+  userDefaultsToList = domain: attrs: map
+    (cmd: "sudo --user=${escapeShellArg config.system.primaryUser} -- ${cmd}")
```

Comment:
Is there a reason you use `sudo -u ${user}` for `launchctl` but `sudo --user=${user}` here?

Reply:
No reason at all. I’ve moved to `--user=` across the board.

Reply:
It looks like `launchctl asuser $(id -u ${user}) sudo --user=${user}` will be necessary here

On a new machine:

```
ssh root@machine darwin-rebuild switch
user defaults...
+ sudo --user=enzime -- defaults write com.apple.dock persistent-apps '...'
2025-05-12 13:10:15.274 defaults[32224:588660] Could not write domain com.apple.dock; exiting
```

Reply:
Done 👍